test/extended/router/http2.go

```diff
@@ -143,39 +208,58 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 				useHTTP2Transport: false,
 			}}
 
-			for i, tc := range testCases {
-				testConfig := fmt.Sprintf("%+v", tc)
-				e2e.Logf("[test #%d/%d]: config: %s", i+1, len(testCases), testConfig)
+			// If we cannot resolve then we're not going
+			// to make a connection, so assert that lookup
+			// succeeds for each route. DNS can take a
+			// long time to rollout for the new subdomain.
+			for _, tc := range testCases {
+				err := wait.PollImmediate(3*time.Second, 20*time.Minute, func() (bool, error) {
+					host := tc.route + "." + shardedDomain
+					addrs, err := net.LookupHost(host)
+					if err != nil {
+						e2e.Logf("host lookup error: %v, retrying...", err)
+						return false, nil
+					}
+					e2e.Logf("host %q now resolves as %+v", host, addrs)
+					return true, nil
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}
+
+			// Shard is using a namespace selector so
+			// label the test namespace to match.
+			err = oc.AsAdmin().Run("label").Args("namespace", oc.Namespace(), "type="+oc.Namespace()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
 
-				// check readiness probe is accessible
-				urlTester := url.NewTester(oc.AdminKubeClient(), oc.KubeFramework().Namespace.Name).WithErrorPassthrough(true)
-				defer urlTester.Close()
-				hostname := getHostnameForRoute(oc, tc.route)
-				urlTester.Within(30*time.Second, url.Expect("GET", "https://"+hostname+"/healthz").Through(hostname).SkipTLSVerification().HasStatusCode(200))
+			for i, tc := range testCases {
+				host := tc.route + "." + shardedDomain
+				e2e.Logf("[test #%d/%d]: GET route: %s", i+1, len(testCases), host)
 
 				var resp *http.Response
-				client := makeHTTPClient(tc.useHTTP2Transport, http2ClientTimeout)
 
-				o.Expect(wait.Poll(time.Second, 30*time.Second, func() (bool, error) {
-					var err error
-					url := "https://" + hostname
-					resp, err = client.Get(url)
+				o.Expect(wait.Poll(3*time.Second, 15*time.Minute, func() (bool, error) {
```

Comment:
Do you think 5 minutes should be enough to wait for a failure? The tests take a pretty long time already to deliver an outcome.

Reply:
I think it is something we could monitor and adjust as we see this go through CI on a daily basis. I would rather not make this flake a lot to begin with.